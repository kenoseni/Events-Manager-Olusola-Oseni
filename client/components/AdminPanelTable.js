import React, { Component } from 'react';

class AdminPanelTable extends Component {
  constructor(props) {
    super(props);
    this.upgradeUserToAdmin = this.upgradeUserToAdmin.bind(this);
  }

  upgradeUserToAdmin(e) {
    e.preventDefault();
    this.props.upgradeUserToAdmin(this.props.user.id);
  }
  render() {
    const { user, i, userDetails } = this.props;
    const { events, centers } = this.props.user;
    return (
      <tr id="users">
        <th scope="row">{i}</th>
        <td>{user.id}</td>
        <td>{user.firstname}</td>
        <td>{user.lastname}</td>
        <td>{user.isAdmin.toString()}</td>
        <td>{user.role}</td>
        <td>{user.email}</td>
        <td id="upgradeuser">
          {!user.isAdmin &&
            userDetails.userrole === 'superadmin' && (
              <button
                id="upgradeuserbutton"
                onClick={this.upgradeUserToAdmin}
                type="button"
                className="btn btn-primary btn-sm my-0"
              >
                UPGRADE
              </button>
            )}
        </td>
      </tr>
    );
  }
}
export default AdminPanelTable;
