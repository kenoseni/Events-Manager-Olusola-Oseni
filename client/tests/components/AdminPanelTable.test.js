import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import Connected, { AdminPanels } from '../../components/AdminPanels';
import AdminPanelTable from '../../components/AdminPanelTable';

configure({ adapter: new Adapter() });
const e = {
  preventDefault() {}
}
describe('<AdminPanelTableList />', () => {
  const mockupgradeUserToAdmin = jest.fn()
  let props = {
    user: {
      events: {},
      centers: {},
      isAdmin: true
    },
    userDetails: {
      userrole: 'superadmin'
    },
  }
  const wrapper = shallow(<AdminPanelTable
    upgradeUserToAdmin={mockupgradeUserToAdmin}
    {...props} />
  )
  it('should render a <AdminPanelTable /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call the upgradeUserToAdmin once', () => {
    const spy = sinon.spy(wrapper.instance(), 'upgradeUserToAdmin')
    wrapper.instance().upgradeUserToAdmin(e)
    expect(mockupgradeUserToAdmin).toHaveBeenCalledTimes(1)
  });
  it('it should render the Upgrade button if user is superadmin', () => {
    expect(wrapper.find('td').exists()).toBeTruthy()
  });
});
