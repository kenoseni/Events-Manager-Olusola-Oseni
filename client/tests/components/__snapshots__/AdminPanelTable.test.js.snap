// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPanelTableList /> should render a <AdminPanelTable /> component 1`] = `
<tr
  id="users"
>
  <th
    scope="row"
  />
  <td />
  <td />
  <td />
  <td>
    true
  </td>
  <td />
  <td />
  <td
    id="upgradeuser"
  />
</tr>
`;
